### PR TITLE
console: migrate HTML helpers, route hooks helpers and version utils to TypeScript

### PR DESCRIPTION
**Pull request by ryands17**
### Affected components

- [x] Console

### Related Issues
https://github.com/hasura/graphql-engine/issues/4314#issue-595358014

### Solution and Design

- Converted the componens `HTML helpers`, `Route Hooks Helpers` and `Version Utils` to their respective TypeScript modules.

#### Breaking changes

- [x] No Breaking changes

### CODE REVIEW

File: console/src/helpers/Html.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+ * Wrapper component containing HTML metadata and boilerplate tags.
+ * Used in server-side code only to wrap the string output of the
+ * rendered route component.
+ *
+ * The only thing this component doesn't (and can't) include is the
+ * HTML doctype declaration, which is added to the rendered output
+ * by the server.js file.
+ */
+
+type IAssets = {
```
Comment (beerose) [inline console/src/helpers/Html.tsx:14]:
```suggestion
type Assets = {
```

As we stick to not prefixing type aliases or interfaces with `I`.

File: console/src/helpers/Html.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  [key in 'styles' | 'javascript']: {
```
Comment (beerose) [inline console/src/helpers/Html.tsx:15]: What about:
```ts
type Assets = Record<'styles' | 'javascript', Record<string, string>>;
```
Nested index signatures may be hard to read.

File: console/src/helpers/versionUtils.ts [hunk 3]
```diff
@@ -10,25 +10,32 @@ const featureLaunchVersions = {
   [READ_ONLY_RUN_SQL_QUERIES]: 'v1.1.0',
 };
 
-export const checkValidServerVersion = version => {
+type IFeature = keyof typeof featureLaunchVersions;
```
Comment (beerose) [inline console/src/helpers/versionUtils.ts:13]:
```suggestion
type Feature = keyof typeof featureLaunchVersions;
```

File: console/src/helpers/versionUtils.ts [hunk 4]
```diff
@@ -10,25 +10,32 @@ const featureLaunchVersions = {
...
+
+type IFeaturesCompatibility = {
```
Comment (beerose) [inline console/src/helpers/versionUtils.ts:15]:
```suggestion
type FeaturesCompatibility = {
```

File: console/src/helpers/Html.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  const { assets } = props;
+  const head = Helmet.rewind();
+  return (
+    <html lang="en-US">
+      <head>
+        <link rel="icon" type="image/png" href="/rstatic/favicon_green.png" />
+        {Object.keys(assets.styles).map((style, key) => (
+          <link
+            href={assets.styles[style]}
+            // eslint-disable-next-line react/no-array-index-key
```
Comment (beerose) [inline console/src/helpers/Html.tsx:36]: Could you add it to the eslint config and disable for all?

File: console/src/helpers/Html.tsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+type HtmlProps = {
+  assets: IAssets;
+  component: React.ReactNode;
+  baseDomain: string;
+};
+
+export default function Html(props: HtmlProps) {
+  const { assets } = props;
+  const head = Helmet.rewind();
```
Comment (beerose) [inline console/src/helpers/Html.tsx:28]: This constant seems not to be used anywhere.
```suggestion
```

File: console/src/helpers/Html.tsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+  };
+};
+
+type HtmlProps = {
+  assets: IAssets;
+  component: React.ReactNode;
+  baseDomain: string;
+};
+
+export default function Html(props: HtmlProps) {
```
Comment (beerose) [inline console/src/helpers/Html.tsx:26]: We usually try use `const` instead `function` (in newer code), so to keep it consistent, could you change it to:
```ts
const Html = (...) => { ... }

export default Html;
```

And also as it's function component we can do:
`export const Html: React.FC<HtmlProps> = props => { ... }`

File: console/bin/server.babel.js [hunk 8]
```diff
@@ -6,7 +6,10 @@ const babelrc = fs.readFileSync('.babelrc');
 let config;
 
 try {
-  config = JSON.parse(babelrc);
+  config = {
+    ...JSON.parse(babelrc),
+    extensions: ['.ts', '.tsx', '.js', '.jsx'],
```
Comment (beerose) [inline console/bin/server.babel.js:11]: 🙌

**Comment by ryands17** [comment]
Hey @beerose :wave: 
Updated the PR with the above suggestions :)

File: console/src/helpers/makeRouteHooksSafe.ts [hunk 9]
```diff
@@ -1,28 +1,32 @@
 import { createRoutes } from 'react-router/lib/RouteUtils';
+import { PlainRoute, EnterHook } from 'react-router';
 
 // Wrap the hooks so they don't fire if they're called before
 // the store is initialised. This only happens when doing the first
 // client render of a route that has an onEnter hook
-function makeHooksSafe(routes, store) {
+
+function makeHooksSafe(routes: PlainRoute[] | PlainRoute, store: any): any {
```
Comment (beerose) [inline console/src/helpers/makeRouteHooksSafe.ts:8]:
```suggestion
function makeHooksSafe(routes: PlainRoute[] | PlainRoute, store: unknown): unknown {
```

File: console/src/helpers/makeRouteHooksSafe.ts [hunk 10]
```diff
@@ -1,28 +1,32 @@
 import { createRoutes } from 'react-router/lib/RouteUtils';
+import { PlainRoute, EnterHook } from 'react-router';
 
 // Wrap the hooks so they don't fire if they're called before
 // the store is initialised. This only happens when doing the first
 // client render of a route that has an onEnter hook
-function makeHooksSafe(routes, store) {
+
+function makeHooksSafe(routes: PlainRoute[] | PlainRoute, store: unknown): unknown {
```
Comment (beerose) [inline console/src/helpers/makeRouteHooksSafe.ts:8]:
```suggestion
function makeHooksSafe(routes: PlainRoute[] | PlainRoute, store: any): unknown {
```

**Pull request merged**